scality_sproxyd_client/sproxyd_client.py

```diff
@@ -254,3 +260,57 @@ def handle_200_or_206(response):
         }
 
         return self._do_http('get_object', handlers, 'GET', name, headers)
+
+    def put_object(self, name, body, user_metadata=None):
+        """Connect to sproxyd and put an object."""
+
+        # The Content-Length is automatically set to the correct value.
+        # See `httplib.HTTPConnection.request`
+        headers = {}
+        if user_metadata:
+            headers['x-scal-usermd'] = base64.b64encode(pickle.dumps(user_metadata))
+
+        handlers = {
+            200: lambda _: None,
+        }
+
+        return self._do_http('put_object', handlers, 'PUT', name,
+                             headers=headers, body=body)
+
+    def get_http_conn_for_put(self, name, headers=None):
+        full_url = self.sproxyd_urls.next() + urllib.quote(name)
+        url_pieces = urlparse.urlparse(full_url)
+        conn_pool = self.http_pools.connection_from_host(url_pieces.hostname,
+                                                         url_pieces.port,
+                                                         url_pieces.scheme)
+        timeout_obj = conn_pool.timeout.clone()
+
+        conn = None
+        try:
+            # Request a connection from the queue.
+            conn = conn_pool._get_conn()
+            conn_pool.num_requests += 1
```

Comment:
https://github.com/shazow/urllib3/blob/master/urllib3/connectionpool.py#L333

https://github.com/shazow/urllib3/blob/master/urllib3/connectionpool.py#L210

I guess one could take a connection from the pool but not make a request with that connection.